docs/modules/ROOT/pages/dataformats.adoc

```diff
@@ -476,4 +484,53 @@ DS.Formats.write(payload, "application/csv", params)
 'Christopher'|'Marlow'|'(987)654-3210'
 ------------------------
 
+## Java Objects
+
+### `read`
+
+Converts POJO to JSON format using https://github.com/FasterXML/jackson[Jackson ObjectMapper].
+
+The following read parameters are supported:
+
+[%header, cols=3*a]
+|===
+|Parameter
+|Description
+|Default value
+
+|`DateFormat`
+|Converts POJO date / time fields to JSON strings using specified date format. See https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html[SimpleDateFormat] for details.
+|`yyyy-MM-dd'T'HH:mm:ss.SSSZ`
+
+|===
+
+### `write`
+
+Converts JSON objects to Java POJOs using https://github.com/FasterXML/jackson[Jackson ObjectMapper].
+
+The following write parameters are supported:
+
+[%header, cols=3*a]
+|===
+|Parameter
+|Description
+|Default value
+
+|`DateFormat`
+|Converts POJO date / time fields to JSON strings using specified date format. See https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html[SimpleDateFormat] for details.
+|`yyyy-MM-dd'T'HH:mm:ss.SSSZ`
+
+|`OutputClass`
+|Produces an instance of specified class. If parameter is not set, the following conversion rules are used:
+
+- Objects -> `java.util.HashMap`
+- Arrays -> `java.util.ArrayList`
+- String values -> `java.lang.String`
+- Boolean values -> `java.lang.Boolean`
+- Numerical values -> `java.lang.Number`
+
+|
+
+|===
 
+IMPORTANT: It is not recommended to call `DS.Formats.write(...)` directly in the DataSonnet code, because it will produce a String representation of a POJO, i.e. `pojo.toString()`. Normally POJO outputs are expected to be the end results of the transformations and are controlled by the appropriate xref:headers.adoc[headers] at the beginning of the transformation code.
```

Comment:
I think I know what this is saying but I'm honestly not sure. I believe it only applies to the java plugin? Also, it seems to be written for java developers and definitely needs clarifying for non-java developers.

Actually, even beyond this, it feels like something that should just not be allowed (in the sense that attempting it should throw a clear error that it doesn't make sense), we should figure out how to do that cleanly